components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -472,16 +472,22 @@ private RolesV2GetResponse filterRolesByAttributes(Node node, Integer count, Int
             LOG.debug(String.format("Filtering roles from search filter: %s", searchFilter));
         }
         List<Role> roles;
+        int roleCount;
+        List<RoleV2> scimRoles;
         try {
             roles = roleManagementService.getRoles(searchFilter, count, startIndex, sortBy, sortOrder, tenantDomain,
                     requiredAttributes);
+            scimRoles = getScimRolesList(roles, requiredAttributes);
+            roleCount = roleManagementService.getRolesCount(searchFilter, tenantDomain);
+            if (roleCount == 0) {
+                roleCount = scimRoles.size();
+            }
```

Comment:
Do we need this?

Reply:
Same reason for adding https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/9cb325ec207e5cd2664d4493b6182122fb15eacf/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java#L552-L555